x/lockup/keeper/lock.go

```diff
@@ -77,26 +77,27 @@ func (k Keeper) deleteLockRefs(ctx sdk.Context, lockRefPrefix []byte, lock types
 	return nil
 }
 
-func (k Keeper) addSyntheticLockRefs(ctx sdk.Context, lockRefPrefix []byte, lock types.PeriodLock) error {
-	refKeys, err := syntheticLockRefKeys(lock)
+// XXX
+func (k Keeper) addSyntheticLockRefs(ctx sdk.Context, lockRefPrefix []byte, synthLock types.SyntheticLock) error {
+	refKeys, err := syntheticLockRefKeys(synthLock)
```

Comment:
Ok so we're now generally changing the flow from adding synthetic lockrefs from a lock for a synth lock, to instead directly be adding them to a mutated lock.

I like this design =)

Would be nice to later re-use a lot of this code